File: python/p4/p4.py
#!/usr/bin/python

def isPalindrome(n):
    strN= str(n)
    if  strN == strN[::-1]: #Same number backwards
        return True
    else:
        return False

init=100
max=1000
j=init
k=init
highest = 0
while j < max:
    while k < max:
        current = j*k
        if isPalindrome(current) and (j*k > highest) :
            highest = j*k
        k=k+1
    j=j+1
    k = init
print highest


            
    
    

File: c_c++/p4/p4.cpp
//A palindromic number reads the same both ways.
//The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
//Find the largest palindrome made from the product of two 3-digit numbers.

#include <iostream>
using namespace std;

//abccba=a*100000+b*10000+c*1000+c*100+b*10+a*1
bool is_palindrome(int number) {
	int tmp=number, inverted_number=0;

	    while (tmp)
	    {
	        inverted_number = inverted_number*10 + (tmp % 10);
	        tmp /= 10;
	    }
	    return (inverted_number == number ? true : false);
}

int main() {
	auto init = 100;
	auto max = 1000;
	auto j = init;
	auto k = init;
	auto highest = 0;
	while(j< max) {
		while (k < max) {
			auto current = j*k;
			if( is_palindrome(current) && current > highest) {
					highest = current;
			}
			k= k+1;
		}
	j = j+1;
	k = init;
	}

	cout << "Number is " << highest << endl;
}


